src/OrchardCore.Modules/OrchardCore.Settings/Services/RecipeEnvironmentSiteNameProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.Recipes.Services;
+
+namespace OrchardCore.Settings
+{
+    public class RecipeEnvironmentSiteNameProvider : IRecipeEnvironmentProvider
+    {
+        private readonly ISiteService _siteService;
+
+        public RecipeEnvironmentSiteNameProvider(ISiteService siteService)
+        {
+            _siteService = siteService;
+        }
+
+        public async Task SetEnvironmentAsync(IDictionary<string, object> environment)
+        {
+            // When these have already been set by another provider, do not reset them.
```

Comment:
Hmm, just not sure if a module depending on OC.Settings want to override the value but have its provider registered later, but yes weird to override the site name